### PR TITLE
0.5.2 optimization

### PR DESCRIPTION
**Pull request by vikasprashar99**
Optimized code for following: -

1. rendercarticon
2. add, remove helper function

@Rahul-Vijay and @Rajat012k Please look for any issues/bugs and merge if not.

### CODE REVIEW

File: lib/screens/common/animatedCard.dart [hunk 1]
```diff
@@ -42,49 +43,49 @@ class _AnimatedCardState extends State<AnimatedCard>
...
+  //     q = double.parse(
+  //         "${widget.product.details[0].quantity.allowedQuantities[index].value}");
+  //     p = widget.product.details[0].price * q;
+  //   }
 
-    addToCart
-        ? newCart.addToCart(widget.product, index, p, q)
-        : newCart.removeFromCart(widget.product, index, p, q);
-  }
+  //   addToCart
```
Comment (Rahul-Vijay) [inline lib/screens/common/animatedCard.dart:85]: We can remove the comments.?

File: lib/screens/common/common_functions.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+///
+/// @fileoverview common functions file : common functions used at multiple
+///  files.
+///
+
+import 'package:mvp/models/newCart.dart';
+import 'package:mvp/models/storeProducts.dart';
+
+class helperFunctions {
+  static void helper(
```
Comment (Rahul-Vijay) [inline lib/screens/common/common_functions.dart:15]: how can we make this generic.? What I mean is, currently we have hard-coded all the `quantity metrics`, we should make it dynamic ? We can have a generic class ?

File: lib/screens/productsNew/details.dart [hunk 3]
```diff
@@ -75,44 +76,44 @@ class _ProductDetailsState extends State<ProductDetails> {
...
-            "Pc" ||
-        widget.p.details[0].quantity.allowedQuantities[index].metric == "Kgs" ||
-        widget.p.details[0].quantity.allowedQuantities[index].metric == "Ltr" ||
-        widget.p.details[0].quantity.allowedQuantities[index].metric ==
-            "Pack") {
-      q = double.parse(
-          "${widget.p.details[0].quantity.allowedQuantities[index].value}");
-      p = widget.p.details[0].price * q;
-    }
+  //   if (widget.p.details[0].quantity.allowedQuantities[index].metric == "Kg") {
```
Comment (Rahul-Vijay) [inline lib/screens/productsNew/details.dart:86]: we can remove all the comments here

**Review by Rahul-Vijay — CHANGES_REQUESTED** [review]
- There is a bug. Steps to reproduce - 
Go to mainLanding page and add one of the bestsellers. Close the page, then go to some other page. Then come back and re-open the added bestseller and try to add or subtract, it doesn't work. 
There is no error in console.

**Comment by vikasprashar99** [comment]
> - There is a bug. Steps to reproduce - 
> Go to mainLanding page and add one of the bestsellers. Close the page, then go to some other page. Then come back and re-open the added bestseller and try to add or subtract, it doesn't work. 
> There is no error in console. 

This was the issue I was taking of , it was before also.

**Comment by Rahul-Vijay** [comment]
> > * There is a bug. Steps to reproduce -
> >   Go to mainLanding page and add one of the bestsellers. Close the page, then go to some other page. Then come back and re-open the added bestseller and try to add or subtract, it doesn't work.
> >   There is no error in console.
> 
> This was the issue I was talking of , it was before also.

Check #90 , you can fix that bug in this PR itself

**Comment by Rahul-Vijay** [comment]
re work this PR @Rajat012k & @vikasprashar99, pull from 0.5.2 and resolve conflicts.

**Review by Rahul-Vijay — APPROVED** [review]

**Pull request merged**